WalletWasabi.Tests/UnitTests/JsonConverters/MoneyBtcJsonConverterTests.cs

```diff
@@ -0,0 +1,177 @@
+			}
+		}
+
+		static void AssertBothDeserialize(string jsonToken)
+		{
+			string json = $$"""{"Name": "Little Book of Calm", "Price": {{jsonToken}} }""";
+
+			TestProduct? product1 = JsonConvertOld.DeserializeObject<TestProduct>(json);
+			TestProduct? product2 = JsonConvertNew.Deserialize<TestProduct>(json);
+
+			// Value equality.
+			Assert.Equal(product1, product2);
+		}
+
+		static void AssertDeserializeFailure<TException>(string jsonToken)
+			where TException : Exception
+		{
+			string json = $$"""{"Name": "Little Book of Calm", "Price": {{jsonToken}} }""";
+
+			Assert.Throws<TException>(() => JsonConvertOld.DeserializeObject<TestProduct>(json));
+			Assert.Throws<TException>(() => JsonConvertNew.Deserialize<TestProduct>(json));
+		}
+
+		static string S(string s)
+			=> $"\"{s}\"";
+	}
+
+	/// <summary>
+	/// Asserts that object <paramref name="o"/> is serialized to the same JSON by both Newtonsoft library and STJ library.
+	/// </summary>
+	/// <returns>JSON representation of <paramref name="o"/>.</returns>
+	private static string AssertSerializedEqually<T>(T o)
+	{
+		string newtonsoftJson = JsonConvertOld.SerializeObject(o);
+		string stjJson = JsonConvertNew.Serialize(o);
+
+		Assert.NotNull(newtonsoftJson);
+		Assert.NotNull(stjJson);
+		Assert.Equal(newtonsoftJson, stjJson);
+
+		return stjJson;
+	}
+
+	/// <summary>
+	/// Record for testing deserialization of <see cref="Money"/>.
+	/// </summary>
+	private record TestProduct
+	{
+		public required string Name { get; init; }
+
+		[Newtonsoft.Json.JsonProperty(PropertyName = nameof(Price))]
+		[Newtonsoft.Json.JsonConverter(typeof(MoneyBtcJsonConverter))]
+		[System.Text.Json.Serialization.JsonConverter(typeof(MoneyBtcJsonConverterNg))]
+		[System.Text.Json.Serialization.JsonPropertyName(nameof(Price))]
+		public Money? Price { get; init; }
+	}
+
+	/// <summary>
+	/// Record with various attributes for both STJ and Newtonsoft.
+	/// </summary>
+	private record TestData
+	{
+		[Newtonsoft.Json.JsonProperty(PropertyName = nameof(Half))]
+		[Newtonsoft.Json.JsonConverter(typeof(MoneyBtcJsonConverter))]
+		[System.Text.Json.Serialization.JsonConverter(typeof(MoneyBtcJsonConverterNg))]
+		[System.Text.Json.Serialization.JsonPropertyName(nameof(Half))]
+		public Money Half { get; set; } = Money.Coins(0.500m);
+
+		[System.Text.Json.Serialization.JsonConverter(typeof(MoneyBtcJsonConverterNg))]
+		[System.Text.Json.Serialization.JsonPropertyName(nameof(One))]
+		[Newtonsoft.Json.JsonProperty(PropertyName = nameof(One))]
+		[Newtonsoft.Json.JsonConverter(typeof(MoneyBtcJsonConverter))]
+		public Money One { get; set; } = Money.Coins(1m);
+
+		[System.Text.Json.Serialization.JsonConverter(typeof(MoneyBtcJsonConverterNg))]
+		[System.Text.Json.Serialization.JsonPropertyName(nameof(Zeros))]
+		[Newtonsoft.Json.JsonProperty(PropertyName = nameof(Zeros))]
+		[Newtonsoft.Json.JsonConverter(typeof(MoneyBtcJsonConverter))]
+		public Money Zeros { get; set; } = Money.Coins(0.000001m);
```

Comment:
I'm not sure about this `Zeros`. 
Is this supposed to be low amount of BTC?
Or why not call it `Zero` and assign `Money.Zero` to it?